lib/posthog/feature_flags.rb

```diff
@@ -42,96 +77,288 @@ def is_feature_enabled(key, distinct_id, default_result = false, groups = {})
+          logger.debug "Successfully computed flag remotely: #{key} -> #{response}"
+        rescue StandardError => e
+          logger.error "Error computing flag remotely: #{e}"
+          response = default_result
         end
-      if feature_flag['is_simple_flag']
-        return is_simple_flag_enabled(key, distinct_id, flag_rollout_pctg)
+      end
+
+      return response
+    end
+
+    def get_all_flags(distinct_id, groups = {}, person_properties = {}, group_properties = {})
+      # make sure they're loaded on first run
+      load_feature_flags
+
+      response = {}
+      fallback_to_decide = @feature_flags.empty?
+
+      @feature_flags.each do |flag|
+        begin
+          response[flag['key']] = _compute_flag_locally(flag, distinct_id, groups, person_properties, group_properties)
+        rescue InconclusiveMatchError => e
+          fallback_to_decide = true
+        rescue StandardError => e
+          logger.error "Error computing flag locally: #{e}."
+          fallback_to_decide = true
+        end
+      end
+
+      if fallback_to_decide
+        begin
+          flags = get_feature_variants(distinct_id, groups, person_properties, group_properties)
+          response = {**response, **flags}
+        rescue StandardError => e
+          logger.error "Error computing flag remotely: #{e}"
+        end
+      end
+      return response
+    end
+
+    def shutdown_poller()
+      @task.shutdown
+    end
+
+    # Class methods
+
+    def self.match_property(property, property_values)
+      # only looks for matches where key exists in property_values
+      # doesn't support operator is_not_set
+
+      key = property['key']
+      value = property['value']
+      operator = property['operator'] || 'exact'
+
+      if !property_values.has_key?(key)
+        raise InconclusiveMatchError.new("Property #{key} not found in property_values")
+      elsif operator == 'is_not_set'
+        raise InconclusiveMatchError.new("Operator is_not_set not supported")
+      end
+
+      override_value = property_values[key]
+
+      if operator == 'exact'
+        if value.is_a?(Array)
+          return value.include?(override_value)
+        end
+        return value == override_value
+      elsif operator == 'is_not'
+        if value.is_a?(Array)
+          return !value.include?(override_value)
+        end
+        return value != override_value
+      elsif operator == 'is_set'
+        return property_values.has_key?(key)
+      elsif operator == 'icontains'
+        return override_value.to_s.downcase.include?(value.to_s.downcase)
+      elsif operator == 'not_icontains'
+        return !override_value.to_s.downcase.include?(value.to_s.downcase)
+      elsif operator == 'regex'
+        return is_valid_regex(value.to_s) && !Regexp.new(value.to_s).match(override_value.to_s).nil?
+      elsif operator == 'not_regex'
+        return is_valid_regex(value.to_s) && Regexp.new(value.to_s).match(override_value.to_s).nil?
```

Comment:
Note: Regex implementations are slightly different in every language. This might need further testing down the line.